pkg/telemetry/phonehome/campaign.go

```diff
@@ -0,0 +1,53 @@
+package phonehome
+
+import (
+	"strings"
+)
+
+// APICallCampaignCriterion defines a criterion for an API interception of a telemetry
+// campaign. Requests parameters need to match all fields for the request to
+// be tracked. Any request matches empty criterion.
+type APICallCampaignCriterion struct {
+	UserAgents     []string          `json:"user_agents,omitempty"`
+	PathPatterns   []string          `json:"path_patterns,omitempty"`
+	Methods        []string          `json:"methods,omitempty"`
+	Codes          []int32           `json:"codes,omitempty"`
+	HeaderPatterns map[string]string `json:"header_patterns,omitempty"`
```

Comment:
We need to document this? Is key also a pattern

Reply:
Changed to `map[string]Pattern`.